cmd/instance_pool_create.go

```diff
@@ -0,0 +1,227 @@
+func (o *instancePoolCreateItemOutput) toText()  { outputText(o) }
+func (o *instancePoolCreateItemOutput) toTable() { outputTable(o) }
+
+var instancePoolCreateCmd = &cobra.Command{
+	Use:     "create <name>",
+	Short:   "Create an instance pool",
+	Aliases: gCreateAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return cmd.Usage()
+		}
+
+		description, err := cmd.Flags().GetString("description")
+		if err != nil {
+			return err
+		}
+
+		zoneName, err := cmd.Flags().GetString("zone")
+		if err != nil {
+			return err
+		}
+
+		if zoneName == "" {
+			zoneName = gCurrentAccount.DefaultZone
+		}
+
+		zone, err := getZoneByName(zoneName)
+		if err != nil {
+			return err
+		}
+
+		so, err := cmd.Flags().GetString("service-offering")
+		if err != nil {
+			return err
+		}
+
+		servOffering, err := getServiceOfferingByName(so)
+		if err != nil {
+			return err
+		}
+
+		templateFilterCmd, err := cmd.Flags().GetString("template-filter")
+		if err != nil {
+			return err
+		}
+		templateFilter, err := validateTemplateFilter(templateFilterCmd)
+		if err != nil {
+			return err
+		}
+
+		templateName, err := cmd.Flags().GetString("template")
+		if err != nil {
+			return err
+		}
+
+		if templateName == "" {
+			templateName = gCurrentAccount.DefaultTemplate
+		}
+
+		template, err := getTemplateByName(zone.ID, templateName, templateFilter)
+		if err != nil {
+			return err
+		}
+
+		keypair, err := cmd.Flags().GetString("keypair")
+		if err != nil {
+			return err
+		}
+
+		if keypair == "" {
+			keypair = gCurrentAccount.DefaultSSHKey
+		}
+
+		size, err := cmd.Flags().GetInt("size")
+		if err != nil {
+			return err
+		}
+
+		sg, err := cmd.Flags().GetStringSlice("security-group")
+		if err != nil {
+			return err
+		}
+
+		sgs, err := getSecurityGroups(sg)
+		if err != nil {
+			return err
+		}
+
+		aff, err := cmd.Flags().GetStringSlice("anti-affinity-group")
+		if err != nil {
+			return err
+		}
+
+		affs, err := getAffinityGroup(aff)
+		if err != nil {
+			return err
+		}
+
+		priv, err := cmd.Flags().GetStringSlice("privnet")
+		if err != nil {
+			return err
+		}
+
+		privs, err := getPrivnetList(priv, zone.ID)
+		if err != nil {
+			return err
+		}
+
+		//It use asyncTasks to have spinner when user exec this command
+		r := asyncTasks([]task{task{
+			egoscale.CreateInstancePool{
+				Name:              args[0],
+				Description:       description,
+				ZoneID:            zone.ID,
+				ServiceofferingID: servOffering.ID,
+				TemplateID:        template.ID,
+				Keypair:           keypair,
+				Size:              size,
+				SecuritygroupIDs:  sgs,
+				AffinitygroupIDs:  affs,
+				NetworkIDs:        privs,
+			},
+			fmt.Sprintf("Create instance pool %q", args[0]),
+		}})
+		errs := filterErrors(r)
+		if len(errs) > 0 {
+			return errs[0]
+		}
+		pool := r[0].resp.(*egoscale.CreateInstancePoolResponse)
+		o, err := formatInstancePoolCreateItemOutput(pool, zone)
+		return output(o, err)
```

Comment:
The recently  global flag `--quiet|-Q` has been introduced for users who don't want any non-vital output to be displayed (for example when running in a script): please have a look in the other commands and implement the same check with `if !gQuiet { ... }`

Reply:
fixed and also in update